test/gtest/common/test_watchdog.cc

```diff
@@ -72,23 +72,30 @@ UCS_TEST_F(test_watchdog, watchdog_signal) {
 }
 
 UCS_TEST_F(test_watchdog, watchdog_timeout) {
-    double timeout;
-    char *p;
+    double timeout, sleep_time;
+    char *gtest_timeout, *gtest_sleep_time;
 
     /* This test can not be run with the other tests
      * because it terminates testing due to timeout
      */
-    p = getenv("WATCHDOG_GTEST_TIMEOUT_");
-    if (p == NULL) {
+    gtest_timeout = getenv("WATCHDOG_GTEST_TIMEOUT_");
+    if (gtest_timeout == NULL) {
         UCS_TEST_SKIP_R("WATCHDOG_GTEST_TIMEOUT_ is not set");
     }
-    ASSERT_TRUE(p != NULL);
+    ASSERT_TRUE(gtest_timeout != NULL);
 
-    timeout = atof(p);
+    gtest_sleep_time = getenv("WATCHDOG_GTEST_SLEEP_TIME_");
+    if (gtest_sleep_time == NULL) {
+        UCS_TEST_SKIP_R("WATCHDOG_GTEST_SLEEP_TIME_ is not set");
+    }
+    ASSERT_TRUE(gtest_sleep_time != NULL);
```

Comment:
can this assert be fired? it is checked few lines above

Reply:
I guess @hoopoepg  means that you just checked that `gtest_sleep_time == NULL`  on line 88

Reply:
thanks @hoopoepg @brminich got it. fixed here and in the place from where the code was copied.